corehq/motech/repeaters/models.py

```diff
@@ -1198,7 +1199,13 @@ def is_success(resp):
                                    RECORD_CANCELLED_STATE)  # Don't retry
 
 
-def format_response(response) -> Optional[str]:
+@overload
+def format_response(response: Any) -> None:
+    ...
+@overload
+def format_response(response: Response) -> str:
+    ...
+def format_response(response):
```

Comment:
When I wrote this, I was thinking, "I bet @millerdev has an opinion on this."

Neat? Or horrifying?

Reply:
I have to confess, I don't love it. I think a docstring would work just as well if you're doing it for documentation purposes. Your IDE (if it's smart enough) can analyze the method and tell you it returns `None` or `str` without the annotations.

I've been thinking about type annotations a bit lately (had a brief interaction with TypeScript), and I'm still asking why people want them and like them so much? I just don't get the hype. I personally find them to make code overly verbose (harder to read) and don't find them to be very useful. If and when they get out of sync with what the code actually does, what then?

One possible reason is nice auto-complete suggestions in your IDE. (Although as I've already mentioned, in this case an IDE could provide all of information without the annotations.) Auto-complete suggestions are handy when you're not familiar with the interface of an object you're using, but it seems like a bit of a crutch. If you're writing code that matters you've probably also got tests, and tests should quickly highlight method name typos. If you need to explore and learn about an API, read the docs or even the source code of thing you're using to learn how it works.

/rant

Apparently I did have an opinion. I hope you're not offended by my dismissive attitude. I'm happy to talk about it more, probably better in person, if feel like I've missed something. It's possible I need to work harder on understanding tyem to fully arrive, but I'm not there yet.

Reply:
> I hope you're not offended by my dismissive attitude.

You'd have to put in a lot more effort to offend me! :joy: 

> I have to confess, I don't love it.

I have to confess, this is not completely unexpected. It's the reason it's in a commit all on its own. :grin: 

Some still-valid points in that rant.

Reply:
406ff20